### PR TITLE
Issues/#90 sha3 foundation

### PR DESCRIPTION
**Pull request by r-soltani**
This PR provides the foundation for SHA3 hashing algorithm.
The hash related files are placed in the src/eqb/ folder. 
The wrapper class is called sha3.h , and it provides basic functionality to create a hash.
This PR does not have a fully functioning test case for sha3.

### CODE REVIEW

File: src/test/crypto_tests.cpp [hunk 1]
```diff
@@ -50,7 +51,10 @@ void TestVector(const Hasher &h, const In &in, const Out &out) {
         BOOST_CHECK(hash == out);
     }
 }
-
```
Comment (ealefad) [inline src/test/crypto_tests.cpp]: Why is there a red line? Was it an empty line? Maybe we should keep it?

File: src/test/crypto_tests.cpp [hunk 2]
```diff
@@ -214,6 +218,28 @@ std::string LongTestString(void) {
...
+//    TestSHA3("abcdbcdecdefdefgefghfghighijhijkijkljklmklmnlmnomnopnopq",
+//               "45d3b367a6904e6e8d502ee04999a7c27647f91fa845d456525fd352ae3d7371");
+//    TestSHA3("For this sample, this 63-byte string will be used as input data",
+//               "5f323aee7801af6f179257f97cde82b40c1b223bb0b05db960151ea29cf01e41");
+//    TestSHA3("This is exactly 64 bytes long, not counting the terminating byte",
+//               "2fd5854e44708f641c67704a46e5514fee6f49c7f413959244e6970ba611ca2e");
+//    TestSHA3("As Bitcoin relies on 80 byte header hashes, we want to have an example for that.",
+//               "d014c33f665c1a7cbf66149112d4cef1d437179d2cb75ecca6b106c31bca21da");
+//}
+#endif // END_BUILD
```
Comment (macterra) [inline src/test/crypto_tests.cpp]: Is it safe to assume that this file contains the only original code in the PR (not counting changes to VS project files)?

File: src/test/crypto_tests.cpp (hunk 1 above)
Comment (r-soltani) [reply src/test/crypto_tests.cpp]: The test file has been changed to the original until the SHA3 tests are complete.

File: src/test/crypto_tests.cpp (hunk 2 above)
Comment (r-soltani) [reply src/test/crypto_tests.cpp]: The source file has been changed to the original until the SHA3 tests are complete.

**Review by macterra — APPROVED** [review]

**Review by ealefad — APPROVED** [review]

File: build_msvc/libequibit_crypto/libequibit_crypto.vcxproj [hunk 3]
```diff
@@ -159,6 +159,27 @@
...
+    <ClInclude Include="..\..\src\eqb\sha3\SimpleFIPS202.h" />
+    <ClInclude Include="..\..\src\eqb\sha3\SnP-Relaned.h" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="..\..\src\eqb\sha3\KeccakP-1600-64.macros" />
+    <None Include="..\..\src\eqb\sha3\KeccakP-1600-unrolling.macros" />
+    <None Include="..\..\src\eqb\sha3\KeccakSponge.inc" />
   </ItemGroup>
   <Import Label="configTarget" Project="..\common.vcxproj" />
-</Project>
```
Comment (KiarashN) [inline build_msvc/libequibit_crypto/libequibit_crypto.vcxproj]: Please don't remove CR

Comment (r-soltani) [reply build_msvc/libequibit_crypto/libequibit_crypto.vcxproj]: fixed

**Pull request merged**